service_management/app/models/analysis/events/Analysis.scala

```diff
@@ -173,3 +184,35 @@ object AnalysisCollection extends WithDateTimeFormatters {
     Json.writes[AnalysisCollection]
 
 }
+
+case class SampleCreated(
+    id: Option[EventId],
+    eventDate: Option[DateTime],
+    registeredBy: Option[ActorId],
+    registeredDate: Option[DateTime],
+    objectId: Option[ObjectUUID],
+    sampleObjectId: Option[ObjectUUID]
+) extends AnalysisEvent {
+  val partOf: Option[EventId]         = None
+  val note: Option[String]            = None
+  val responsible: Option[ActorId]    = None
+  val administrator: Option[ActorId]  = None
+  val updatedBy: Option[ActorId]      = None
+  val updatedDate: Option[DateTime]   = None
+  val completedBy: Option[ActorId]    = None
+  val completedDate: Option[DateTime] = None
+  val analysisTypeId                  = SampleCreated.sampleEventTypeId
+}
+
+object SampleCreated extends WithDateTimeFormatters {
+  val sampleEventTypeId: AnalysisTypeId = AnalysisTypeId.empty
+
+  val discriminator = "SampleCreated"
+
+  def reads(implicit r: Reads[Analysis]): Reads[SampleCreated] =
```

Comment:
As it stands, the _input_ argument has the wrong type.

When above comment is applied, this needs to change into an `implicit val reads: Reads[SampleCreated]` that also handles the `discriminator` as handled in the current code in the `AnalysisEvent` companion object.